mlx/backend/metal/primitives.cpp

```diff
@@ -627,4 +627,8 @@ void Transpose::eval_gpu(const std::vector<array>& inputs, array& out) {
   eval(inputs, out);
 }
 
+void QRF::eval_gpu(const std::vector<array>& inputs, array& out) {
+  eval(inputs, out);
```

Comment:
I think it's better to throw here with a "Metal QR factorization not yet implemented" rather than implicitly go to the CPU implementation. That way people know what they are getting.